core/content/proxy/content_store.go

```diff
@@ -18,26 +18,49 @@ package proxy
 
 import (
 	"context"
+	"fmt"
 	"io"
 
 	contentapi "github.com/containerd/containerd/v2/api/services/content/v1"
 	"github.com/containerd/containerd/v2/core/content"
 	"github.com/containerd/containerd/v2/protobuf"
 	protobuftypes "github.com/containerd/containerd/v2/protobuf/types"
 	"github.com/containerd/errdefs"
+	"github.com/containerd/ttrpc"
 	digest "github.com/opencontainers/go-digest"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
 )
 
 type proxyContentStore struct {
-	client contentapi.ContentClient
+	// client is the rpc content client
+	// NOTE: ttrpc is used because it is the smaller interface shared with grpc
+	client contentapi.TTRPCContentClient
 }
 
 // NewContentStore returns a new content store which communicates over a GRPC
 // connection using the containerd content GRPC API.
-func NewContentStore(client contentapi.ContentClient) content.Store {
-	return &proxyContentStore{
-		client: client,
+func NewContentStore(client any) content.Store {
```

Comment:
Can we use a type union `[T ContentClient | ClientConnInterface]` instead of `any`?

Reply:
I tried this before, not sure if it just disallows interfaces. Error like ` cannot use google.golang.org/grpc.ClientConnInterface in union (google.golang.org/grpc.ClientConnInterface contains methods)`